--- lfs-bootscripts-20111017/lfs/lib/services/init-functions-ORIG	2011-10-21 20:30:16.000000000 -0700
+++ lfs-bootscripts-20111017/lfs/lib/services/init-functions	2012-02-18 13:46:12.093976105 -0800
@@ -487,26 +487,31 @@
     local pidlist
 
     # Process arguments
-    while true; do
+    local args
+    while [ $# -ne 0 ] ; do
       case "${1}" in
-
           -p)
+              args="${args} ${1}"
               pidfile="${2}"
               shift 2
               ;;
+           *)
+              args="${args} ${1}"
+              shift 1
+              ;;
       esac
    done
 
    if [ -z "${pidfile}" ]; then
-      pidlist=`pidofproc -p "${pidfile}" $@`
+      pidlist=`pidofproc $args`
    else
-      pidlist=`pidofproc $@`
+      pidlist=`pidofproc -p "${pidfile}" $@`
    fi
 
    # Trim trailing blanks
    pidlist=`echo "${pidlist}" | sed -r 's/ +$//'`
 
-   base="${1##*/}"
+   base="${args##*/}"
 
    if [ -n "${pidlist}" ]; then
       echo -e "${INFO}${base} is running with Process" \
@@ -742,4 +747,26 @@
    return 0
 }
 
+################################################################################
+# is_dir_empty()                                                               #
+#                                                                              #
+# Purpose: Checks if a directory has no entries (files, subdirs, etc)          #
+#                                                                              #
+# Inputs: Accepts a single string which represents a path                      #
+#                                                                              #
+# Return values:                                                               #
+#       0 - No entries                                                         #
+#       1 - Enries present                                                     #
+#     254 - Invalid directory path given                                       #
+#     255 - No argument given                                                  #
+#                                                                              #
+################################################################################
+is_dir_empty()
+{
+   local dir=$1
+   if [ -z $dir ] ; then echo "$dir unspecified ; aborting" ; return 255 ; fi
+   if [ ! -d $dir ] ; then echo "$dir is not a directory; aborting" ; return 254 ; fi
+   if [ 0 -eq $(/bin/ls -1 $dir/* 2> /dev/null | wc -l) ] ; then return 0 ; else return 1 ; fi
+}
+
 # End /lib/lsb/init-functions
